lib/archethic/contracts/interpreter/scope.ex

```diff
@@ -19,27 +21,109 @@ defmodule Archethic.Contracts.Interpreter.Scope do
   @doc """
   Create a new nested scope
   """
-  @spec create(list(String.t())) :: :ok
-  def create(scope_hierarchy) do
+  @spec create() :: :ok
+  def create() do
+    current_context = get_current_context()
+    current_scope_hierarchy = get_context_scope_hierarchy(current_context)
+    ref = new_ref()
+
+    Process.put(
+      :scope,
+      put_in(Process.get(:scope), [current_context] ++ current_scope_hierarchy ++ [ref], %{})
+    )
+
+    Process.put(
+      :scope,
+      update_in(Process.get(:scope), [current_context, "scope_hierarchy"], &(&1 ++ [ref]))
+    )
+
+    :ok
+  end
+
+  @doc """
+  Create new context
+  """
+  @spec create_context() :: :ok
+  def create_context() do
+    context_ref = new_ref()
+
+    new_context = %{
+      "scope_hierarchy" => []
+    }
+
+    # add context to scope and update context list
+    new_scope =
+      Process.get(:scope)
+      |> Map.put(context_ref, new_context)
+      |> Map.update!("context_list", &[context_ref | &1])
+
+    Process.put(:scope, new_scope)
+
+    :ok
+  end
+
+  @doc """
+  Leave a scope by removing it from current context's scope hierarchy and deleting its content
+  """
+  @spec leave_scope() :: :ok
+  def leave_scope() do
```

Comment:
Here you only delete the ref in the scope hierarchy list. You could also delete the ref in the nested scope map, it would behave a bit like a garbage collector